FIX: Database fix for empty category name - move childs and rebuild paths

Root-Core opened the pull request:
Fixes #191

The current code base now handles empty categories, but it does not fix existing issues.
This PR fixes the DB and makes content accessible again.

Disclaimer: No AI was used in this pull request.

Root-Core committed:
DB: Fix for empty category name / move childs / rebuild paths

nzbdav-dev commented on backend/Database/Migrations/20251220004937_Fix-Empty-Categories.cs:
```diff
@@ -0,0 +1,72 @@
+            // Previous duplicates will be removed due to 'ON DELETE CASCADE'
+            migrationBuilder.Sql(@"
+                DELETE FROM DavItems WHERE Name = ''
+            ");
```
I think we want,

```
migrationBuilder.Sql(@"
  UPDATE DavItems
  SET Name = COALESCE(
      (SELECT ConfigValue FROM ConfigItems WHERE ConfigName = 'api.manual-category'),
      'uncategorized'
  )
  WHERE Name = ''
    AND ParentId = '00000000-0000-0000-0000-000000000002'
");
```


This will rename the empty-name category folder to whatever is configured in the `api.manual-category` setting.
Then `BuildFullPath` should rebuild all the paths.

Root-Core replied:
I think this could result in a collision if the category already exists. This approach cleanly moves all the items and removes the unnecessary empty category.

nzbdav-dev commented on backend/Database/Migrations/20251220004937_Fix-Empty-Categories.cs:
```diff
@@ -0,0 +1,72 @@
+
+#nullable disable
+
+namespace NzbWebDAV.Database.Migrations
```
I wonder if this migration is worth the complexity. 
I'm worried about edge cases that could end up making things worse for someone's db.

Would replacing the entire Up()` method in this migration with something like this be a safer bet?

```
delete from QueueItems where Category = '';
delete from HistoryItems where Category ='';
delete from DavItems where Name = '' and ParentId = '00000000-0000-0000-0000-000000000002';
```


This has very predictable results, and would "fix" the problem well-enough for those that encountered the https://github.com/nzbdav-dev/nzbdav/issues/191 bug

Root-Core commented on backend/Config/ConfigManager.cs:
```diff
@@ -56,16 +56,16 @@ public void UpdateValues(List<ConfigItem> configItems)
-        var mountDir = StringUtil.EmptyToNull(GetConfigValue("rclone.mount-dir"))
-                       ?? StringUtil.EmptyToNull(Environment.GetEnvironmentVariable("MOUNT_DIR"))
+        var mountDir = GetConfigValue("rclone.mount-dir")
+                       ?? EnvironmentUtil.GetVariable("MOUNT_DIR")
```
> `Environment.GetEnvironmentVariable` and `EnvironmentUtil.GetVariable` are slightly different.

Yes, I realised that after creating the PR. I suppose it can be fixed without compromising the improved interface.

> Maybe we should rename the latter to `EnvironmentUtil.GetRequiredVariable` to make the distinction clearer. Anyway, the change here that updates to use `EnvironmentUtil.GetVariable` prevents the `"/mnt/nzbdav"` default from ever being reached, since it will throw an exception if MOUNT_DIR is not configured.

I had also considered something along these lines.

> I'd leave this entire file as-is tbh, the refactors here seem unrelated to the goal of the pull-request, which is to address cases where an empty category is specified in the "AddFile" and "AddUrl" apis.

The PR started with fixing the empty string problem. The improved API should prevent these mistakes from happening.. it later shifted to also fixing the DB.

Root-Core replied:
I have modified the `GetVariable()` function to accept a fallback. Please let me know if this is an acceptable solution for you.

Root-Core commented:
@nzbdav-dev Should I rebase on the `dev` branch or wait until you merge it into `main`?

I see, that you also addressed `GetVariable()` issue. Have a look at my solution and let me know which one you prefer.
My is a bit less verbose and more generic, yours involve less refactoring.

Root-Core commented:
> * While I appreciate the refactor, I personally think `StringUtil.EmptyToNull(...)` is a little more clear on what it does versus the `.OrNull()` extension.

One last try. :)
I renamed it to `ToNullIfEmpty()`, as it casts to `null`. This is more clear and the `To` part  indicates the type change.
  
> * And I'm generally a fan of extension methods :) The repo is [full of them](https://github.com/nzbdav-dev/nzbdav/tree/0d2b675b62ab7fb26e19cc4dc42be0e3d38516ac/backend/Extensions).

You are right, I missed that.. I moved the extension into the existing `StringExtensions.cs`.

> * These two comments are the ones mostly giving me pause:
>       
>     * https://github.com/nzbdav-dev/nzbdav/pull/216/changes#r2643537774
>     * https://github.com/nzbdav-dev/nzbdav/pull/216/changes#r2643587160
> 
> At this stage of the project, I'm inclined to keep pr acceptance to those that are minimal and single-purpose

I understand your point and normally I would try keeping it lean, but the PR changed a bit in it's goals.
See my replies to your comments. I think the PR is in a good state and the migration is pretty polished.
